### PR TITLE
First work

### PR DESCRIPTION
**Pull request by JayH5**
- Add code from praekelt/seed-stack
- Add misc files
- Fix lint errors and warnings
- Add README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,89 @@
...
+* [praekeltfoundation/marathon](https://github.com/praekeltfoundation/puppet-marathon)
+
+Then there are some 3rd party modules:
+* [camptocamp/openssl](https://forge.puppetlabs.com/camptocamp/openssl)
+* [deric/mesos](https://forge.puppetlabs.com/deric/mesos)
+* [deric/zookeeper](https://forge.puppetlabs.com/deric/zookeeper)
+* [garethr/docker](https://forge.puppetlabs.com/garethr/docker)
+* [gdhbashton/consul_template](https://forge.puppetlabs.com/gdhbashton/consul_template)
+* [KyleAnderson/consul](https://forge.puppetlabs.com/KyleAnderson/consul)
+* [stankevich/python](https://forge.puppetlabs.com/stankevich/python)
```
Comment (jerith) [inline README.md]: Do we still need `stankevich/python`? I think that was for doing venv stuff before we moved to debs and upstart.

File: metadata.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    {
+      "name": "puppet",
+      "version_requirement": ">=3.4.0 < 4.0.0"
+    }
+  ],
+  "operatingsystem_support": [
+    {
+      "operatingsystem": "Ubuntu",
+      "operatingsystemrelease": [
+        "12.04",
```
Comment (jerith) [inline metadata.json]: Do we want to support Precise here? Are we able to?

Comment (JayH5) [inline metadata.json]: I don't see any reason why it wouldn't work... but I guess we're not actually testing/supporting it so maybe I should remove it...

File: README.md (hunk 1 above)
Comment (JayH5) [inline README.md]: We don't need it, thanks.

File: metadata.json (hunk 2 above)
Comment (jerith) [inline metadata.json]: If we keep it, we should test it -- I suspect we'll run into problems with old versions of deps and such, though. (I don't object to supporting it, I just don't see any particularly compelling reason to put in the effort.)

**Pull request merged**